instrumentation/grpc-1.6/library/src/main/java/io/opentelemetry/instrumentation/grpc/v1_6/GrpcAttributesExtractor.java

```diff
@@ -5,18 +5,31 @@
 
 package io.opentelemetry.instrumentation.grpc.v1_6;
 
+import static io.opentelemetry.instrumentation.grpc.v1_6.CapturedGrpcMetadataUtil.lowercase;
+import static io.opentelemetry.instrumentation.grpc.v1_6.CapturedGrpcMetadataUtil.requestAttributeKey;
+
 import io.grpc.Status;
 import io.opentelemetry.api.common.AttributesBuilder;
 import io.opentelemetry.context.Context;
 import io.opentelemetry.instrumentation.api.instrumenter.AttributesExtractor;
 import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
+import java.util.List;
 import javax.annotation.Nullable;
 
 final class GrpcAttributesExtractor implements AttributesExtractor<GrpcRequest, Status> {
+  private final GrpcRpcAttributesGetter getter;
+  private final List<String> capturedRequestMetadata;
+
+  GrpcAttributesExtractor(
+      GrpcRpcAttributesGetter getter, List<String> requestMetadataValuesToCapture) {
+    this.getter = getter;
+    this.capturedRequestMetadata = lowercase(requestMetadataValuesToCapture);
+  }
+
   @Override
   public void onStart(
       AttributesBuilder attributes, Context parentContext, GrpcRequest grpcRequest) {
-    // No request attributes
+    // Request attributes captured on request end.
```

Comment:
is it possible to capture attributes on start so they can be used for sampling?

Reply:
@trask Capture them only at start or at both start and end?

Reply:
can they be different at those two times?

Reply:
@trask Not sure what you mean by that

Reply:
sorry, I mean, if you capture the request metadata in onStart, is there a reason to capture them in onEnd also?

Reply:
@trask So do you think it should only be on start?

Reply:
if we don't know of any reason to also be on end, then let's start with "only on start"

Reply:
@trask 
Status capture has to be on end because it doesn't exist at start.
I moved the metadata capture to start though